packages/sdk38/src/lcdclient.spec.ts

```diff
@@ -0,0 +1,857 @@
+              amount: "7654321",
+            },
+          ],
+        },
+      };
+
+      const fee: StdFee = {
+        amount: [
+          {
+            amount: "5000",
+            denom: "ucosm",
+          },
+        ],
+        gas: "890000",
+      };
+
+      const client = new LcdClient(wasmd.endpoint);
+      const { account_number, sequence } = (await client.authAccounts(address1)).result.value;
+
+      const signBytes = makeSignBytes([msg1, msg2], fee, wasmd.chainId, memo, account_number, sequence);
+      const signature1 = await account1.sign(signBytes);
+      const signedTx = {
+        msg: [msg1, msg2],
+        fee: fee,
+        memo: memo,
+        signatures: [signature1],
+      };
+      const postResult = await client.postTx(signedTx);
+      expect(postResult.code).toBeUndefined();
+    });
+
+    it("can send multiple messages with multiple signatures", async () => {
+      pendingWithoutWasmd();
+      const account1 = await Secp256k1Pen.fromMnemonic(faucet.mnemonic, makeCosmoshubPath(0));
+      const account2 = await Secp256k1Pen.fromMnemonic(faucet.mnemonic, makeCosmoshubPath(1));
+      const address1 = rawSecp256k1PubkeyToAddress(account1.pubkey, "cosmos");
+      const address2 = rawSecp256k1PubkeyToAddress(account2.pubkey, "cosmos");
+
+      const memo = "My first contract on chain";
+      const msg1: MsgSend = {
+        type: "cosmos-sdk/MsgSend",
+        value: {
+          from_address: address1,
+          to_address: defaultRecipientAddress,
+          amount: [
+            {
+              denom: "ucosm",
+              amount: "1234567",
+            },
+          ],
+        },
+      };
+      const msg2: MsgSend = {
+        type: "cosmos-sdk/MsgSend",
+        value: {
+          from_address: address2,
+          to_address: defaultRecipientAddress,
+          amount: [
+            {
+              denom: "ucosm",
+              amount: "7654321",
+            },
+          ],
+        },
+      };
+
+      const fee: StdFee = {
+        amount: [
+          {
+            amount: "5000",
+            denom: "ucosm",
+          },
+        ],
+        gas: "890000",
+      };
+
+      const client = new LcdClient(wasmd.endpoint);
+      const { account_number: an1, sequence: sequence1 } = (await client.authAccounts(address1)).result.value;
+      const { account_number: an2, sequence: sequence2 } = (await client.authAccounts(address2)).result.value;
+
+      const signBytes1 = makeSignBytes([msg2, msg1], fee, wasmd.chainId, memo, an1, sequence1);
+      const signBytes2 = makeSignBytes([msg2, msg1], fee, wasmd.chainId, memo, an2, sequence2);
+      const signature1 = await account1.sign(signBytes1);
+      const signature2 = await account2.sign(signBytes2);
+      const signedTx = {
+        msg: [msg2, msg1],
+        fee: fee,
+        memo: memo,
+        signatures: [signature2, signature1],
+      };
+      const postResult = await client.postTx(signedTx);
```

Comment:
Okay, so you can technically post with LcdClient (like rest client), but then we wrap it with the `SigningClient` to make this all much more easy to handle?

Reply:
Every LCD client can broadcast. It is a trivial API call open to everyone. The hard part is the signing, which is where the differentiation is. So this is not exactly read-only vs. read-write.